reconcile/utils/saasherder.py

```diff
@@ -388,13 +388,42 @@ def _process_template(self, options):
                 return None, None
 
             if "IMAGE_TAG" not in consolidated_parameters:
-                template_parameters = template.get('parameters')
-                if template_parameters is not None:
-                    for template_parameter in template_parameters:
-                        if template_parameter['name'] == 'IMAGE_TAG':
-                            # add IMAGE_TAG only if it is required
-                            image_tag = commit_sha[:hash_length]
-                            consolidated_parameters['IMAGE_TAG'] = image_tag
+                for template_parameter in template.get("parameters", {}):
+                    if template_parameter["name"] == "IMAGE_TAG":
+                        # add IMAGE_TAG only if it is required
+                        image_tag = commit_sha[:hash_length]
+                        consolidated_parameters["IMAGE_TAG"] = image_tag
+
+            # Do this in a separate loop, since it relies on IMAGE_TAG already
+            # being calculated.
+            # This should almost never already be set in
+            # consolidated_parameters, but just in case...
+            if "REPO_DIGEST" not in consolidated_parameters:
+                for template_parameter in template.get("parameters", {}):
+                    if template_parameter["name"] == "REPO_DIGEST":
+                        try:
+                            logging.info("Generating REPO_DIGEST.")
+                            import docker
+                            docker_client = docker.from_env()
```

Comment:
Are we sure that saas deployer has pull credentials in place to ANY images that can ever be templated?

Reply:
Maybe what we have to do is to keep IMAGE_TAG unchanged and introduce a new IMAGE_DIGEST tag that will trigger this special behavior.

Reply:
On top of that, we need to make sure we have at least the quay.io/app-sre pull credentials in the saas-deployer job env.

Reply:
Probably not. Nor do I suppose we can count on that being the case across the board, since the push could technically happen anywhere, not necessarily in chain with this saas deployer run.

I'm going to say we should defer that problem to future, if/when we actually encounter it. In the meantime, if someone tries to request `REPO_DIGEST` for a private image, this will blow up, which is good. (I.e. we don't want it to silently succeed with some bogus value.)

Reply:
> Maybe what we have to do is to keep IMAGE_TAG unchanged and introduce a new IMAGE_DIGEST tag that will trigger this special behavior.

That's exactly what I've done (except `s/IMAGE_DIGEST/REPO_DIGEST/`).

> On top of that, we need to make sure we have at least the quay.io/app-sre pull credentials in the saas-deployer job env.

I mean, we _can_, but perhaps that's future work? I can make a note (where?) that this currently only works for public images.

Reply:
Update here: I've switched to using `sretoolbox.container.Image` and funneled in the same `image_auth` used elsewhere. So this should be good as far as creds already exist. And I believe that also puts the onus on the client to make sure they have the right creds in place if necessary, right?